pixee/codemodder-java
# Adding integration test to ReplaceDefaultHttpClientCodemod

Overview:
- ReplaceDefaultHttpClientCodemod looks for the use of `DefaultHttpClient()` and replace it with `HttpClientBuilder`
-  In order to test the behavior after the code transformation, a test project is added with an endpoint that makes a request to https://restcountries.com/ using `DefaultHttpClient`.  The endpoint receive a country name and returns the capital.


What Changed:
- `ReplaceDefaultHttpClientCodemodIntegrationTest` class added 
-  test project is added

How to test:
- in order to run the integration test locally, follow README in core-codemods/src/test/java/io/codemodder/codemods/integration/README.md
- Currently the test is marked as disabled, `@Disabled` annotation should be removed to run the test
Evidence:
<img width="1118" alt="Screenshot 2023-10-25 at 10 50 12" src="https://github.com/pixee/codemodder-java/assets/36873975/eab59f81-8550-4ce9-80af-bf5f3a854c67">
